qiskit_experiments/curve_analysis/base_curve_analysis.py

```diff
@@ -214,7 +207,7 @@ def _default_options(cls) -> Options:
 
         options.plotter = CurvePlotter(MplDrawer())
         options.plot_raw_data = False
-        options.return_fit_parameters = True
+        options.return_fit_parameters = False
```

Comment:
Yeah, ideally I would like to deprecate accessing results by number entirely. I also mentioned up above that the order of artifacts will change when you save and load a composite experiment: https://github.com/Qiskit-Extensions/qiskit-experiments/pull/1342#issuecomment-1920926420.

In this case, I can shift simple indices by 1, but `analysis_results()` accepts slices and we'd have to introduce a bunch of code checking if the slice includes 0 and how to shift it. I'm not sure what the best solution is—the simplest is just to give a warning when users try to use numerical indices.

Reply:
I don't think you want to shift all indices because only CurveAnalysis indices changed. That's most things but some like tomography do not use CurveAnalysis and would not have shifted.

Maybe we should not accept `int` for `ExperimentData.artifacts()`?

Reply:
By the way, based on #1385 it seems like we are treating the moving of the fit parameters from analysis results to artifacts as a minor non-core change. Does that seem right?

Reply:
Oh I think I misunderstood you. I thought you wanted to put all the logic in `expdata.analysis_results()` but it would be easier to just add a placeholder in the actual `AnalysisResultTable`. Would the placeholder be actual data or a blank row? I think if we're going to get rid of numerical indices across the package, we could be more aggressive in making the breaking change now to get users used to the idea of not using them.

Reply:
Yes, I was thinking something mostly blank. It could have a name like  `@Parameters_Placeholder` or something like that and `ExperimentData.analysis_results(0)` could error if the result being returned were `@Parameters_Placeholder`. I think that is the only way the user gets a result directly?

Reply:
Then I think it's better to just put back the original behavior and warn the user if they try to use `analysis_results(0)` or numerical indices. I think it would confuse new users why there's a placeholder there with no useful information. One issue with moving to non-numerical indices is that we don't make the analysis result name very easy to find in the docs or regularized. Some are styled like `Local Readout Mitigator` and some are `d_theta`.